### ISSUE TITLE
[PF2] [FEATURE REQUEST] Allow users to split their spells by spell slots

### ISSUE THREAD

**Issue by yoshisman8** [Drental/fvtt-tokenactionhud#21]
When an actor has a lot of spell slots (Such as a high level caster) and the user doesn't have a particularly large monitor. The list of spells reaches the bottom of the screen and makes it impossible to utilize the spells tab of the HUD. Having an option to split the spell tab into multiple spell slots would be really useful.

Example of a lv6 character having a lot of spells listed and almost reaching the bottom of the screen:
![image](https://user-images.githubusercontent.com/4214215/124974117-6d176580-dffa-11eb-8acb-a0fd51b40477.png)

**Comment by apoapostolov** [comment]
@Drental maybe support for sublevel categories - you point at Spells, then sublevel categories appear for each level, and each subcategory works as a category for the content filter to it.

**Comment by Drental** [comment]
Thinking about it, I hope to find a solution that is not solving vertical size issues by creating horizontal ones.
This one is definitely not straightforward easy.

**Comment by xdy** [comment]
Maybe not expand the spell list until the level is clicked? I.e. something like:
```
Cantrips (9 spells click to expand)
Level 1 (5 spells, click to expand)
Level 2 Augury, Calm Emotions, etc (i.e. full spell list)
Level 3 (6 spells, click to expand)
...etc...
```
Might be worth making 'expand all spell levels by default' into a setting. Possibly even something like 'collapse all levels but [list]'.

**Comment by Larkinabout** [comment]
I think this has been mostly resolved as of 3.0.4 by the addition of a scroll bar for long lists. It avoids expanding the width of the HUD or adding more clicks (albeit by adding more scrolling). It's also quicker to scroll up and down to explore your spells, then it would be to hover over or click each spell level to see what spells exist underneath.

**Comment by Larkinabout** [comment]
Resolved for D&D 5e with [Token Action HUD D&D 5e](https://foundryvtt.com/packages/token-action-hud-dnd5e) with scrolling content. Collapsible/expandable section may be a future development, but not high up on the priority list at the moment.
